kore/src/Kore/Step/Simplification/Defined.hs

```diff
@@ -0,0 +1,28 @@
+{-|
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+module Kore.Step.Simplification.Defined
+    ( simplify
+    ) where
+
+import Prelude.Kore
+
+import Kore.Internal.OrPattern
+    ( OrPattern
+    )
+import qualified Kore.Internal.OrPattern as OrPattern
+import Kore.Internal.TermLike
+
+-- | Marks an 'OrPattern' resulted from simplification
+-- as 'Defined'. Does not do any actual simplification.
+simplify
+    :: InternalVariable variable
+    => Defined (OrPattern variable)
+    -> OrPattern variable
+simplify Defined { getDefined = defined } =
+    OrPattern.fromTermLike
+    $ markSimplified
+    $ mkDefined
+    $ OrPattern.toTermLike defined
```

Comment:
This will incorrectly mark the `predicate` and `substitution` as defined, which effectively deletes them. We want to map `mkDefined` over only the `TermLike` terms.